### PR TITLE
Add missing documentation to behavior components

### PR DESCRIPTION
**Pull request by ben-dodd-dev**
Filled in many missing documentation descriptions from the behaviour components schema. 
- A few descriptions are based on those from www.bedrock.dev or my own testing/experience
- A few values were missing minimum/maximum properties and have been added where necessary

Related to issue #15

### CODE REVIEW

File: source/behavior/entities/format/behaviors/harvest_farm_block.json [hunk 1]
```diff
@@ -10,19 +10,19 @@
...
       "title": "Maximum Seconds Before Search",
-      "description": "UNDOCUMENTED",
+      "description": "The maximum amount of time in seconds that the goal can take before searching for the first harvest block.",
       "$comment": "UNDOCUMENTED",
       "type": "number"
     },
     "search_cooldown_max_seconds": {
       "title": "Search Cooldown Maximum Seconds",
-      "description": "UNDOCUMENTED",
+      "description": "The maximum amount of time in seconds that the goal can take before searching again, after failing to find a a harvest block already.",
```
Comment (DaanV2) [inline source/behavior/entities/format/behaviors/harvest_farm_block.json:19]: Check spelling on this one, the ending gets a bit weird

File: source/behavior/entities/format/components/block_sensor.json [hunk 2]
```diff
@@ -7,10 +7,11 @@
   "properties": {
     "sensor_radius": {
       "type": "integer",
-      "description": "UNDOCUMENTED",
+      "description": "The maximum radius in blocks in which a specified block can be detected.",
       "$comment": "UNDOCUMENTED",
       "title": "Sensor Radius",
-      "minimum": 0
+      "minimum": 0,
+      "maximum": 32
```
Comment (DaanV2) [inline source/behavior/entities/format/components/block_sensor.json:14]: I need to check if this is indeed the range

**Review by DaanV2 — CHANGES_REQUESTED** [review]
Looks good and thanks for the hard work ☺️ saw a single line that's needs to be checked, and two things I need to confirm in Minecraft

File: source/behavior/entities/format/behaviors/defend_trusted_target.json [hunk 3]
```diff
@@ -52,9 +52,11 @@
     },
     "sound_chance": {
       "title": "Sound Chance",
-      "description": "UNDOCUMENTED",
+      "description": "The chance that the `aggro_sound` will play",
       "$comment": "UNDOCUMENTED",
-      "type": "number"
+      "type": "number",
+      "minimum": 0,
+      "maximum": 1
```
[the remark this answers is not in the document]
Comment (mitgobla) [reply source/behavior/entities/format/behaviors/defend_trusted_target.json:59]: Of course. I added the min/max as chances are usually a float between 0 and 1

**Pull request merged**